Guide/src/reference/architecture/openhcl/processes.md

```diff
@@ -0,0 +1,121 @@
+# OpenHCL Processes and Components
+
+This document describes the major software components and processes that make up the OpenHCL paravisor environment.
+
+## Boot Shim (`openhcl_boot`)
+
+The boot shim is the first code that executes in VTL2. It is responsible for early hardware initialization and preparing the environment for the Linux kernel.
+
+**Source code:** [openhcl/openhcl_boot](https://github.com/microsoft/openvmm/tree/main/openhcl/openhcl_boot) | **Docs:** [openhcl_boot rustdoc](https://openvmm.dev/rustdoc/linux/openhcl_boot/index.html)
+
+**Key Responsibilities:**
+
+- **Hardware Initialization:** Sets up CPU state, enables MMU, and configures initial page tables.
+- **Configuration Parsing:** Receives boot parameters from the host that were generated at IGVM build time.
```

Comment:
Maybe its worth mentioning here that it may also perform filtering of the host-provided configuration in CVM scenarios to ensure nothing slips through that would weaken our security boundaries (like say exposing debugging services)